sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -187,15 +187,18 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
 
   test("test different encoding") {
     // scalastyle:off
-    spark.sql(
-      s"""
+    withView("carsTable") {
+      spark.sql(
+        s"""
          |CREATE TEMPORARY TABLE carsTable USING csv
```

Comment:
as you are fixing this test suite, can you replace all `CREATE TEMPORARY TABLE` to `VIEW`?